packages/core/security/core-security-common/src/authentication/authenticated_user.ts

```diff
@@ -65,4 +77,9 @@ export interface AuthenticatedUser extends User {
    * Indicates whether user is an operator.
    */
   operator?: boolean;
+
+  /**
+   * Metadata of the API key that was used to authenticate the user.
+   */
+  api_key?: ApiKey;
```

Comment:
__question:__ @elastic/es-security is there any reason we don't document optional `api_key` property of the [`Authenticate API` response](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-authenticate.html)? Is it something stable enough Kibana can rely on?

Reply:
nope, no good reason not to document -- it's stable enough to rely on 👍

Reply:
I'm not sure how what the `ApiKey` model here is -- the authentication response will not give you back full API key info (i.e. it won't have role descriptors and such), just `id` and unless it's a very old API key a `name` field

Reply:
Good point, it should be `ApiKeyDescriptor` or something along these lines to not conflict with the proper `ApiKey` interface we have elsewhere.

Reply:
Resolved in this commit [731080b](https://github.com/elastic/kibana/pull/204114/commits/731080b08a9c21b4bc7bae66bc8faa6c10defc74)